pyfolio/perf_attrib.py

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright 2017 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import division
+import pandas as pd
+import numpy as np
+
+from pyfolio.utils import print_table
+
+
+def perf_attrib(factor_loadings,
+                factor_returns,
+                strategy_daily_returns,
+                strategy_daily_holdings):
+    """
+    Does performance attribution given risk info.
+
+    Parameters
+    ----------
+    factor_loadings : pd.DataFrame
+        Factor loadings for all days in the date range, with date and ticker as
+        index, and factors as columns.
+        - Example:
+                               momentum  reversal
+            dt         ticker
+            2017-01-01 AAPL   -1.592914  0.852830
+                       TLT     0.184864  0.895534
+                       XOM     0.993160  1.149353
+            2017-01-02 AAPL   -0.140009 -0.524952
+                       TLT    -1.066978  0.185435
+                       XOM    -1.798401  0.761549
+
+    factor_returns : pd.DataFrame
+        Returns by factor, with date as index and factors as columns
+        - Example:
+                        momentum  reversal
+            2017-01-01  0.002779 -0.005453
+            2017-01-02  0.001096  0.010290
+
+    strategy_daily_returns : pd.DataFrame
+        Returns for each day in the date range.
+        - Example:
+                            AAPL       TLT       XOM
```

Comment:
This might not be relevant at all, but I believe TLT and XOM are two of our sector ETFs... @twiecki are we concerned that we mention them in a docstring?